haystack/document_stores/weaviate.py

```diff
@@ -243,6 +262,14 @@ def _convert_weaviate_result_to_document(
                 # weaviate returns already scaled values
                 if score and not scale_score:
                     score = score * 2 - 1
+            elif "distance" in props["_additional"]:
+                score = props["_additional"]["distance"]
+                # Weaviate returns the negative dot product. To make score comparable
+                # to other document stores, we take the negative.
+                if self.similarity == "dot" and score:
+                    score = -1 * score
+                if score and scale_score:
+                    score = self.scale_to_unit_interval(score, self.similarity)
```

Comment:
I think changing this block to the following is more readable:

```suggestion
                if score:
                    # Weaviate returns the negative dot product. To make score comparable
                    # to other document stores, we take the negative.
                    if self.similarity == "dot":
                        score = -1 * score
                    if scale_score:
                        score = self.scale_to_unit_interval(score, self.similarity)
```

Reply:
Agreed.